src/components/Company/PaySchedule/PaySchedule.tsx

```diff
@@ -0,0 +1,362 @@
+  })
+}
+
+interface PayScheduleProps extends CommonComponentInterface {
+  companyId: string
+  defaultValues?: PayScheduleDefaultValues
+}
+
+export const PaySchedule = ({
+  companyId,
+  defaultValues,
+  ...props
+}: PayScheduleProps & BaseComponentInterface) => {
+  useI18n('Company.PaySchedule')
+  return (
+    <BaseComponent {...props}>
+      <Root companyId={companyId} defaultValues={defaultValues}>
+        {props.children}
+      </Root>
+    </BaseComponent>
+  )
+}
+
+const Root = ({ companyId, children, defaultValues }: PayScheduleProps) => {
+  const { baseSubmitHandler, onEvent } = useBase()
+  const [mode, setMode] = useState<MODE>('LIST_PAY_SCHEDULES')
+  const [currentPaySchedule, setCurrentPaySchedule] = useState<
+    (Schemas['Pay-Schedule'] & { version?: string }) | null
+  >(null)
+  const transformedDefaultValues: PayScheduleInputs = {
+    frequency: defaultValues?.frequency ?? 'Every week',
+    anchor_pay_date: defaultValues?.anchor_pay_date
+      ? parseDate(defaultValues.anchor_pay_date)
+      : undefined,
+    anchor_end_of_pay_period: defaultValues?.anchor_end_of_pay_period
+      ? parseDate(defaultValues.anchor_end_of_pay_period)
+      : undefined,
+    day_1: defaultValues?.day_1 ?? undefined,
+    day_2: defaultValues?.day_2 ?? undefined,
+    custom_name: defaultValues?.custom_name ?? '',
+    auto_pilot: defaultValues?.auto_pilot?.toString() ?? '',
+    custom_twice_per_month: 'false',
+  }
+
+  const [payScheduleDraft, setPayScheduleDraft] = useState<PaySchedulePreviewDraft | null>(null)
+  const {
+    data: payPreviewData,
+    error: paySchedulePreviewError,
+    isLoading,
+  } = useGetPaySchedulePreview(
+    companyId,
+    payScheduleDraft as PaySchedulePreviewDraft, // Casting to non-null because we know it's not null from the enabled prop
+    !!payScheduleDraft,
+  )
+
+  const { data: paySchedules } = useGetAllPaySchedules(companyId)
+
+  const createPayScheduleMutation = useCreatePaySchedule()
+  const updatePayScheduleMutation = useUpdatePaySchedule()
+
+  const formMethods = useForm<PayScheduleInputs, unknown, PayScheduleOutputs>({
+    resolver: valibotResolver(PayScheduleSchema),
+    defaultValues: transformedDefaultValues,
+  })
+
+  const allValues = formMethods.watch()
+
+  // Set the custom_twice_per_month value based on the frequency and day_1 and day_2 values as it is not set by the API call
+  useEffect(() => {
+    if (
+      allValues.frequency === 'Twice per month' &&
+      allValues.day_1 === 15 &&
+      allValues.day_2 === 31 &&
+      allValues.custom_twice_per_month === undefined
+    ) {
+      formMethods.setValue('custom_twice_per_month', `false`)
+    } else if (
+      allValues.frequency === 'Twice per month' &&
+      allValues.custom_twice_per_month === undefined
+    ) {
+      formMethods.setValue('custom_twice_per_month', `true`)
+    }
+  }, [
+    allValues.frequency,
+    allValues.day_1,
+    allValues.day_2,
+    formMethods.setValue,
```

Comment:
Nit: you can get rid of this warning

<img width="1043" alt="Screenshot 2025-03-10 at 10 40 48 AM" src="https://github.com/user-attachments/assets/6099e545-3cab-4bf5-8cf9-a0885e91761c" />

if you just do

```
const { setValue } = formMethods

useEffect(() => {
  ...
}, [setValue, ...]
```

(same for the one below)